### PR TITLE
Feature/#27 #28 drive and search

### PR DESCRIPTION
**Pull request by arpi09**
Front end finished. Waiting for API to finish to be able to connect front end with back end.

### CODE REVIEW

File: src/main/resources/webassets/js/custom/drive.js [hunk 1]
```diff
@@ -0,0 +1,19 @@
+var base = base || {};
+base.driveController = (function() {
+    var controller = {
+        load: function() {
```
Comment (jost95) [inline src/main/resources/webassets/js/custom/drive.js]: Arrow function can be used (though watch for use of this in object method, won't bind properly)

File: src/main/resources/webassets/js/custom/drive.js [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+            
+            base.Drive =
```
Comment (jost95) [inline src/main/resources/webassets/js/custom/drive.js]: No right side of assignment

File: src/main/resources/webassets/js/custom/drive.js [hunk 3]
```diff
@@ -0,0 +1,19 @@
+var base = base || {};
+base.driveController = (function() {
+    var controller = {
+        load: function() {
+            
+            base.Drive = 
+            base.rest.putDrive(id, drive).then(function(response) {
```
Comment (jost95) [inline src/main/resources/webassets/js/custom/drive.js]: Arrow function can be used

File: src/main/resources/webassets/js/custom/drive.js [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+                console.log("Driver added");
```
Comment (jost95) [inline src/main/resources/webassets/js/custom/drive.js]: Remove console output

File: src/main/resources/webassets/js/custom/drive.js [hunk 5]
```diff
@@ -0,0 +1,19 @@
...
+            }),
+                base.rest.getDrive().then(function(response) {
+                console.log(response);
```
Comment (jost95) [inline src/main/resources/webassets/js/custom/drive.js]: Remove console output

File: src/main/resources/webassets/js/custom/drive.js [hunk 6]
```diff
@@ -0,0 +1,19 @@
+var base = base || {};
+base.driveController = (function() {
+    var controller = {
+        load: function() {
+            
+            base.Drive = 
+            base.rest.putDrive(id, drive).then(function(response) {
+                console.log("Driver added");
+            }),
+                base.rest.getDrive().then(function(response) {
```
Comment (jost95) [inline src/main/resources/webassets/js/custom/drive.js]: Arrow function can be used

File: src/main/resources/webassets/js/custom/drive.js [hunk 7]
```diff
@@ -0,0 +1,19 @@
...
+                console.log(response);
+            })
+        },
+        initOnLoad: function() {
```
Comment (jost95) [inline src/main/resources/webassets/js/custom/drive.js]: Arrow function can be used

File: src/main/resources/webassets/js/custom/search.js [hunk 8]
```diff
@@ -0,0 +1,14 @@
+var base = base || {};
+base.searchController = (function() {
```
Comment (jost95) [inline src/main/resources/webassets/js/custom/search.js]: Arrow function can be used

File: src/main/resources/webassets/js/custom/search.js [hunk 9]
```diff
@@ -0,0 +1,14 @@
...
+    var controller = {
+        load: function() {
+            base.rest.getDrives().then(function(response) {
```
Comment (jost95) [inline src/main/resources/webassets/js/custom/search.js]: Arrow function can be used

File: src/main/resources/webassets/js/custom/search.js [hunk 10]
```diff
@@ -0,0 +1,14 @@
+var base = base || {};
+base.searchController = (function() {
+    var controller = {
+        load: function() {
```
Comment (jost95) [inline src/main/resources/webassets/js/custom/search.js]: Arrow function can be used

File: src/main/resources/webassets/js/custom/search.js [hunk 11]
```diff
@@ -0,0 +1,14 @@
...
+            base.rest.getDrives().then(function(response) {
+                    console.log(JSON.stringify(response) + "hej");
```
Comment (jost95) [inline src/main/resources/webassets/js/custom/search.js]: Remove console output?

File: src/main/resources/webassets/js/custom/search.js [hunk 12]
```diff
@@ -0,0 +1,14 @@
...
+            });
+        },
+        initOnLoad: function() {
```
Comment (jost95) [inline src/main/resources/webassets/js/custom/search.js]: Arrow function can be used

File: src/main/resources/webassets/scss/custom/main.scss [hunk 13]
```diff
@@ -11,4 +11,108 @@ body {
...
 }
 
-@import "../vendors/bootstrap/bootstrap";
\ No newline at end of file
+@import "../vendors/bootstrap/bootstrap";
+
+
+///////DRIVE CSS//////
+
+.top {
```
Comment (jost95) [inline src/main/resources/webassets/scss/custom/main.scss]: Might be better to use more descriptive class declarations, e.g. drive-top-box

File: src/main/resources/webassets/scss/custom/main.scss [hunk 14]
```diff
@@ -11,4 +11,108 @@ body {
...
+    background-color: white;
```
Comment (jost95) [inline src/main/resources/webassets/scss/custom/main.scss]: Use hashcode

File: src/main/resources/webassets/scss/custom/main.scss [hunk 15]
```diff
@@ -11,4 +11,108 @@ body {
...
+    border-top-left-radius: 10px;
+    border-top-right-radius: 10px;
+}
+
+.mid {
+    background-color: #F8E9A1;
+}
+
+.bot {
+    background-color: #A8D0E6;
```
Comment (jost95) [inline src/main/resources/webassets/scss/custom/main.scss]: Save this color, you're reusing it

File: src/main/resources/webassets/scss/custom/main.scss [hunk 16]
```diff
@@ -11,4 +11,108 @@ body {
...
+    border-bottom-left-radius: 10px;
+    border-bottom-right-radius: 10px;
+}
+
+.circle-grade {
+    width: 120px;
+    height: 120px;
+    border-radius: 50%;
+    font-size: 20px;
+    color: black;
```
Comment (jost95) [inline src/main/resources/webassets/scss/custom/main.scss]: Use hashcode

File: src/main/resources/webassets/templates/drive.html [hunk 17]
```diff
@@ -0,0 +1,87 @@
+<!DOCTYPE html>
```
Comment (jost95) [inline src/main/resources/webassets/templates/drive.html]: The template files should start with <div class="row">... should not include html, head and body tags

File: src/main/resources/webassets/templates/drive.html [hunk 18]
```diff
@@ -0,0 +1,87 @@
...
+                    </div>
+                </div>
+            </div>
+        </div>
+
+
+        <script src="js/custom/rest.js"></script>
+        <script src="js/custom/drive.js"></script>
+        <script type="application/javascript">
+            base.driveController.initOnLoad();
```
Comment (jost95) [inline src/main/resources/webassets/templates/drive.html]: window.base....

File: src/main/resources/webassets/templates/search.html [hunk 19]
```diff
@@ -0,0 +1,100 @@
+<!DOCTYPE html>
```
Comment (jost95) [inline src/main/resources/webassets/templates/search.html]: Remove html, head, body tags

File: src/main/resources/webassets/templates/search.html [hunk 20]
```diff
@@ -0,0 +1,100 @@
...
+                    </div>
+                </div>
+            </div>
+        </div>
+
+
+        <script src="js/custom/rest.js"></script>
+        <script src="js/custom/search.js"></script>
+        <script type="application/javascript">
+            base.searchController.initOnLoad();
```
Comment (jost95) [inline src/main/resources/webassets/templates/search.html]: window.base

**Comment by arpi09** [comment]
Requested changes by Jon hava been completed.

**Pull request merged**